File: Library/Searching/Binary_Search/binarySearch.py
def binarySearch(arr, l, r, item):
    while l<=r:
        mid = l+(r-1) //2
        if arr[mid] == item:
            return mid
        elif arr[mid] < item:
            l = mid+1

        else:
            r = mid-1
    return -1

# Driver Code
arr = [2, 3, 4, 10, 40]
x = 10

# Function call
result = binarySearch(arr, 0, len(arr) - 1, x)

if result != -1:
    print("Element is present at index % d" % result)
else:
    print("Element is not present in array")

File: Library/Searching/Binary_Search/binarySearch.cpp
#include <bits/stdc++.h>
using namespace std;

int binarySearch(int arr[], int n, int start, int end) {
    int mid;
    while(start<end){
        mid = (start+end)/2;
        if(arr[mid]==n){
            return mid;
        }
        else if(arr[mid]<n){
            end = mid-1;
            
        }
        else{
            start = mid+1;
        }
    }
    return -1;

}

int main() {
    int n;
    cin >> n;

    int arr[n];
    for(int i = 0; i<n; i++) {
        cin >> arr[i];
    }
    int item;
    cin >> item;

    int ans = binarySearch(arr, item, 0, n-1);
    if(ans ==-1){
        cout << "item not found" << endl;
    }
    else{
        cout << "item found" << endl;
    }

    return 0;
}

/*
1.  Sample input:
    5
    1 2 3 4 5
    3
    Output: item found

2. Sample input
    5
    1 2 3 4 5
    10
    Output: item not found

*/